http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -608,29 +608,20 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       )
     } yield ()
 
-  def handleCreateAppMessage(msg: CreateAppMessage)(
+  private[monitor] def handleCreateAppMessage(msg: CreateAppMessage)(
     implicit ev: ApplicativeAsk[F, AppContext]
   ): F[Unit] =
     for {
       ctx <- ev.ask
 
-      initialGoogleCall = msg.cluster
-        .traverse(createClusterMessage =>
-          gkeInterp.createCluster(createClusterMessage.clusterId, List(createClusterMessage.defaultNodepoolId), false)
-        )
-      clusterResult <- initialGoogleCall.adaptError {
-        case e =>
-          PubsubKubernetesError(
-            AppError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
-            Some(msg.appId),
-            false,
-            msg.cluster.map(_.defaultNodepoolId),
-            msg.cluster.map(_.clusterId)
+      // kick off cluster creation if necessary
+      createAndPollCluster = msg.cluster
```

Comment:
I think the original code intend to keep some google calls being kicked off synchronous to messaging acking instead of queueing them, is that still true after the refactoring?

Reply:
Yeah I thought about that -- right now everything is asynchronous. I wasn't sure the reasoning behind keeping some things synchronous to pubsub message processing...?

Reply:
reason is if some initial google calls fail, then we can nack the message and pubsub will resend the message.

Reply:
I guess maybe we could do the very first create cluster call sync… but beyond that I don't think we want to resend the message since things could be in a bad state.